app/gui/view/src/project.rs

```diff
@@ -47,7 +47,12 @@ pub mod project_view_top_bar;
 /// A time which must pass since last change of expression of node which is the searcher input
 /// to send `searcher_input_changed` event. The delay ensures we don't needlessly update Component
 /// Browser when user is quickly typing in the expression input.
-const INPUT_CHANGE_DELAY_MS: i32 = 200;
+///
+/// Note: This value is set to zero, which means that the event is processed as fast as possible,
+/// but still at the earliest in the next frame. This means that there is a minimal delay but
+/// other events can be processed in the meantime. This is important since the event can lead to
+/// a lot of processing in the Component Browser and we don't want to block the UI.
+const INPUT_CHANGE_DELAY_MS: i32 = 0;
```

Comment:
I have two questions:
1. How does this timing change fit into the solution? The change in `searcher.rs` fixes the race condition, right?
2. Does this negatively impact performance in some circumstances? The case I'm concerned about is: When the user is typing faster than we can process the changes, do we accumulate work that has to be done (processing states that have already passed) before we can catch up to the current state?

Reply:
I would keep the input delay if possible, precisely for the case described by Kaz (in point 2):

> Does this negatively impact performance in some circumstances? The case I'm concerned about is: When the user is typing faster than we can process the changes, do we accumulate work that has to be done (processing states that have already passed) before we can catch up to the current state?

This is exactly what happened, and that's why the delay was introduced in the first place. We can consider lowering it, but still I'd catch several quick key presses for one update.

Reply:
> How does this timing change fit into the solution? The change in searcher.rs fixes the race condition, right?

Only the combination of changes fixes the race condition, as we (a) need to be sure the CB is correctly refreshed and (b) we have processed all the keys before dropping the CB. 

> Does this negatively impact performance in some circumstances? The case I'm concerned about is: When the user is typing faster than we can process the changes, do we accumulate work that has to be done (processing states that have already passed) before we can catch up to the current state?

No, this does not accumulate work. The nice thing here is, that if the processing of the list takes longer, even this one-frame delay makes sure, that all key presses that occur during heavy load are then processed in the following frame together. So, all key presses during heavy load are collected together in the following frame and processed as one edit event in the frame following that. 

I have not observed any performance regressions during my testing, but we would definitely reconfirm during QA.

Reply:
I don't think this is the only solution for the problem. Instead, when user is typing letters, they are put in a "queue" and component browser refreshes only knowing this queue is empty, or anything like that. 

Anyway, there is a very important info missing here - as stated above, how does it affect performance on a weak hardware? The delay was introduced to improve performance if I remember correctly. Of course, I'd prefer not to have this delay if we can be fast enough, and maybe we are now - but this needs to be checked and confirmed.

Reply:
There is now a solution that better addresses the underlying issue. It appears that changing the `INPUT_CHANGE_DELAY_MS` to zero only coincidentally fixed the issue. The new solution does away with that change and leaves the value untouched.